sql/catalyst/src/main/scala/org/apache/spark/sql/types/Interval.scala

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.types
+
+
+/**
+ * The internal representation of interval type.
+ */
+case class Interval(months: Int, microseconds: Long) extends Serializable
```

Comment:
actually can you make this a java class and move it to unsafe/src/main/java/org/apache/spark/unsafe/types ?

Just leave the fields public so we can easily access them in codegen.